.github/workflows/merge-main.yml

```diff
@@ -125,29 +123,95 @@ jobs:
 
-  deploys-prod:
-    name: PROD Deployments
+  deploys-prod-nats:
+    name: Deploy NATS to PROD
     needs:
       - deploys-test
     environment: prod
     runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up OpenShift CLI
+        uses: redhat-actions/oc-login@v1
+        with:
+          openshift_server_url: ${{ vars.OC_SERVER }}
+          openshift_token: ${{ secrets.OC_TOKEN }}
+          namespace: ${{ vars.OC_NAMESPACE }}
+
+      - name: Add NATS Helm repo
+        run: helm repo add nats https://nats-io.github.io/k8s/helm/charts/
+
+      - name: Check if NATS release exists in PROD
+        run: |
+          RELEASE_NAME=${{ github.event.repository.name }}-prod-nats
+          if helm list -q | grep -q $RELEASE_NAME; then
+            echo "PROD Release $RELEASE_NAME already exists. Skipping install."
+            echo "release_exists=true" >> $GITHUB_ENV
+          else
+            echo "PROD Release $RELEASE_NAME does not exist. Proceeding with install."
+            echo "release_exists=false" >> $GITHUB_ENV
+          fi
+
+      - name: Deploy NATS to PROD using Helm
+        if: env.release_exists == 'false'
+        run: |
+          helm install ${{ github.event.repository.name }}-prod-nats nats/nats \
+            --set config.jetstream.enabled=true \
+            --set config.jetstream.fileStore.pvc.size=200Mi \
+            --set config.jetstream.memoryStore.enabled=true \
+            --set config.jetstream.memoryStore.maxSize=250Mi \
+            --set cluster.enabled=true \
+            --set natsBox.enabled=false \
+            --set persistence.enabled=true \
+            --set persistence.size=200Mi \
+            --set container.merge.resources.requests.cpu=100m \
+            --set container.merge.resources.limits.cpu=200m \
+            --set container.merge.resources.requests.memory=100Mi \
+            --set container.merge.resources.limits.memory=400Mi \
+            --set reloader.enabled=false \
+            --set replicaCount=1
+
+  deploys-prod-backend:
+    name: Deploy Backend to PROD
+    needs:
+      - deploys-prod-nats
+    environment: prod
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: bcgov-nr/action-deployer-openshift@v2.3.0
+        with:
+          file: backend/openshift.deploy.yml
+          oc_namespace: ${{ vars.OC_NAMESPACE }}
+          oc_server: ${{ vars.OC_SERVER }}
+          oc_token: ${{ secrets.OC_TOKEN }}
+          overwrite: true
+          parameters: -p ZONE=prod -p PROMOTE=${{ github.repository }}/backend:test
+            -p NAME=${{ github.event.repository.name }} -p PROMOTE_MIGRATION=${{ github.repository }}/database-migrations:test
+          penetration_test: false
+
+  deploys-prod:
+    name: Deploy Other Components to PROD
+    needs:
+      - deploys-prod-backend
+    environment: prod
+    runs-on: ubuntu-22.04
     strategy:
       matrix:
-        name: [backend, database, frontend, init]
```

Comment:
Why was backend removed from here?   I think we need it to pass in the flyway parameters?  

Also will need to resolve the merge-conflict.

Reply:
Good observation.  In the past, by using the matrix strategy, webeoc would start before the backend.  To fix this, I'm first deploying the backend first (it still has the flyway parameters, see the deploys-prod-backend job in merge-main.yml), and then deploying everything else after.  

That said, I see an issue with this approach (even though it will work).  The issue is that this only works if the database already exists (which is does).  If it doesn't, the backend wouldn't be able to start without the database.  

Also, I should probably do the same for test.  I'll make that fix.

Reply:
I don't see the FLYWAY_LOCATIONS parameter in the backend deploy?  Although it will probably still work as it is the default (I think)

`  deploys-prod-backend:
    name: Deploy Backend to PROD
    needs:
      - deploys-prod-nats
    environment: prod
    runs-on: ubuntu-22.04
    steps:
      - uses: bcgov-nr/action-deployer-openshift@v2.3.0
        with:
          file: backend/openshift.deploy.yml
          oc_namespace: ${{ vars.OC_NAMESPACE }}
          oc_server: ${{ vars.OC_SERVER }}
          oc_token: ${{ secrets.OC_TOKEN }}
          overwrite: true
          parameters: -p ZONE=prod -p PROMOTE=${{ github.repository }}/backend:test
            -p NAME=${{ github.event.repository.name }} -p PROMOTE_MIGRATION=${{ github.repository }}/database-migrations:test
          penetration_test: false`

Reply:
Okay we'll see how it goes, the test environment appears to be working correctly, and we can re-verify prod.